src/Lykke.Service.BlockchainWallets/Workflow/CommandHandlers/EndTransactionHistoryMonitoringCommandHandler.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Lykke.Cqrs;
+using Lykke.Service.BlockchainApi.Client;
+using Lykke.Service.BlockchainWallets.Core;
+using Lykke.Service.BlockchainWallets.Core.Repositories;
+using Lykke.Service.BlockchainWallets.Core.Services;
+using Lykke.Service.BlockchainWallets.Workflow.Commands;
+
+namespace Lykke.Service.BlockchainWallets.Workflow.CommandHandlers
+{
+    [UsedImplicitly]
+    public class EndTransactionHistoryMonitoringCommandHandler
+    {
+        private readonly IBlockchainIntegrationService _blockchainIntegrationService;
+        private readonly IMonitoringSubscriptionRepository _monitoringSubscriptionRepository;
+
+
+        public EndTransactionHistoryMonitoringCommandHandler(
+            IBlockchainIntegrationService blockchainIntegrationService,
+            IMonitoringSubscriptionRepository monitoringSubscriptionRepository)
+        {
+            _blockchainIntegrationService = blockchainIntegrationService;
+            _monitoringSubscriptionRepository = monitoringSubscriptionRepository;
+        }
+
+        [UsedImplicitly]
+        public async Task<CommandHandlingResult> Handle(EndTransactionHistoryMonitoringCommand command, IEventPublisher publisher)
+        {
+            var apiClient = _blockchainIntegrationService.TryGetApiClient(command.BlockchainType);
+
+            if (apiClient == null)
+            {
+                throw new NotSupportedException($"Blockchain type [{command.BlockchainType}] is not supportedю");
+            }
+            
+            
+            const MonitoringSubscriptionType subscriptionType = MonitoringSubscriptionType.TransactionHistory;
+
+            var address = command.Address;
+            var assetId = command.AssetId;
+            var blockchainType = command.BlockchainType;
+
+            
+            await _monitoringSubscriptionRepository.UnregisterWalletSubscriptionAsync
+            (
+                blockchainType: blockchainType,
+                address: address,
+                assetId: assetId,
+                subscriptionType: subscriptionType
+            );
+            
+            if (await _monitoringSubscriptionRepository.WalletSubscriptionsCount(blockchainType, address, subscriptionType) == 0)
```

Comment:
And here

Reply:
As we've discussed - this is count of the address+asset subscriptions in the storage, but in blockchain API we have just address subscription, so for multi asset blockchains we need to call API only when there are no address+asset subscriptions